src/hdmf/common/table.py

```diff
@@ -555,7 +555,7 @@ def create_region(self, **kwargs):
         region = getargs('region', kwargs)
         if isinstance(region, slice):
             if (region.start is not None and region.start < 0) or (region.stop is not None and region.stop > len(self)):
-                msg = 'region slice %s is out of range for this DynamicTable of length ' % (str(region), len(self))
```

Comment:
i'm surprised this wasn't caught by flake8 (or us) before!

Reply:
I'm guessing it was never hit by any code